src/components/TweetList.js

```diff
@@ -11,7 +11,31 @@ const defaultProps = {
 };
 
 function renderTweets(tweets) {
-  return tweets.map(tweet => <Tweet tweet={tweet} key={tweet.id} />);
+
+  tweets.forEach(tweet => {
+    let imageSmall;
+
+    if(tweet['entities']['media'] !== undefined){
+      console.log(tweet['entities']);
+      imageSmall = tweet['entities']['media'][0]['media_url'] + ':small';
+      Object.assign(tweet, {imageSmall: imageSmall});
+    }
+
+    let link;
+
+    if(/http\S*/.test(tweet['text'])){
```

Comment:
This is doing 2 reg ex checks (one in the if and one for the match). You may as well just do the matching and check if the result is null.